nat/src/stateful/allocator_writer.rs

```diff
@@ -1,17 +1,18 @@
 // SPDX-License-Identifier: Apache-2.0
```

Comment:
I'd update the comment to say it might be bigger, in practice it may be the same size because of how Rust can use niches to implement the enum.